File: bci/fakebci.py
import sys
import random


class FakeBCI(object):
    def __init__(self, channels=4):
        self.rand = random.Random(1337)
        self.getdata_count = 0
        self.acquire_count = 0
        self.ret = None
        self.channels = channels
        
    def open(self, port):
        self.port = port
        return 1
        
    def init(self, analog_channels, digital_channels):
        self.init = True
        self.achan = analog_channels
        self.dchan = digital_channels
        return 1
        
    def start(self):
        self.start = True
        return 1
        
    def acquire(self):
        self.acquire_count += 1
        return 1
        
    def getdata(self, samples_cross_channels):
        self.getdata_count += 1
        self.ret = []        
        for i in range(samples_cross_channels):
            self.ret.append(self.rand.randint(1, sys.maxint))
            
        return self.ret
        
        

File: bci/FakeBci.cpp
#include <boost/random/uniform_int_distribution.hpp>
#include <limits>

#include "fakeBci.hpp"

FakeBci::FakeBci()
  : mOpenCount(0), mOpenRet(true), mInitCount(0), mLastChannels(4), mInitRet(true), mChannelsCount(0), mStartCount(0), mStartRet(true),
    mAcquireCount(0), mAcquireRet(1), mGetDataCount(0), mpLastGetData(0), mLastSamples(0),
    mTimestampCount(0), mTimestampRet(-1), mStopCount(0), mStopRet(true), mCloseCount(0),
    mCloseRet(true)
{
  mLastMac[0] = 0xd;
  mLastMac[1] = 0xe;
  mLastMac[2] = 0xa;
  mLastMac[3] = 0xd;
  mLastMac[4] = 0xe;
  mLastMac[5] = 0xd;
}

FakeBci::~FakeBci() {
}

bool FakeBci::open(uint8_t* pMacAddress) {
  BOOST_VERIFY(pMacAddress != 0);	
  mOpenCount++;
  std::copy(pMacAddress, pMacAddress+MAC_ADDRESS_SIZE, mLastMac);
  return mOpenRet;
}

bool FakeBci::init(size_t channels) {
  mInitCount++;
  mLastChannels = channels;
  return mInitRet;
}

size_t FakeBci::channels() {
  mChannelsCount++;
  return mLastChannels;
}

bool FakeBci::start() {
  mStartCount++;
  return mStartRet;
}

size_t FakeBci::acquire() {
  mAcquireCount++;
  return mAcquireRet;
}

void FakeBci::getdata(uint32_t* buffer, size_t samples) {
  BOOST_VERIFY(buffer != 0);
  BOOST_VERIFY(mAcquireRet == samples);
  mGetDataCount++;
  for(size_t i=0; i < samples; i++) {
    for (size_t j=0; j < mLastChannels; j++) {
      boost::random::uniform_int_distribution<> dist(1, std::numeric_limits<int32_t>::max());
      buffer[i+j] = (uint32_t)dist(gen);
    }
  }
  mpLastGetData = buffer;
  mLastSamples = samples;
}

uint64_t FakeBci::timestamp() {
  mTimestampCount++;
  return mTimestampRet;
}

bool FakeBci::stop() {
  mStopCount++;
  return mStopRet;
}

bool FakeBci::close() {
  mCloseCount++;
  return mCloseRet;
}

